File: ServiceBook/ClassLibrary1/Models/Vehicle.cs
using Microsoft.AspNet.Identity.EntityFramework;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceBook.DAL.Models
{
    public class Vehicle
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }
        public string VIN { get; set; }
        public string Identifier { get; set; }
        public int OID { get; set; }
        [ForeignKey("OID")]
        public virtual VehicleOwner VehicleOwner{ get; set; }

        public virtual ICollection<Service> Services { get; set; }

        

    }
}

File: ServiceBook/ServiceBook/Scripts/Service/Models/Vehicle.js
function Vehicle(data)
{
    this.vehicleOwner = new VehicleOwner();
    this.id=null;
    this.vin=null;
    this.identifier=null;
    this.oid=null;
    this.services=null;
    if(data!=null)
    {
        this.id=data.ID;
        this.vin=data.VIN;
        this.identifier=data.Identifier;
        this.oid=data.OID;
        this.services = _.map(data.Services, function (service, index) {
            return new Service(service);

        });
        this.vehicleOwner = new VehicleOwner(data.VehicleOwner);

    }
}

